server/api/recipes.js

```diff
@@ -16,15 +16,46 @@ router.get('/:id', (req, res, next) => {
 });
 
 router.post('/', (req, res, next) => {
-  Recipe.create(req.body)
-    .then((newRecipe) => {
-      res.status(201).json(newRecipe);
+  Recipe.findOrCreate({
+    where: {
+      title: req.body.title,
+      author: req.body.author,
+    },
+    defaults: {
+      recipeUrl: req.body.recipeUrl,
+      imageUrl: req.body.imageUrl,
+      siteName: req.body.siteName,
+      tags: req.body.tags,
+      isFavorite: req.body.isFavorite,
+      numServings: req.body.numServings,
+    },
+  })
+    .spread((newRecipe, isCreated) => {
+      if (isCreated) {
+        const arrIngredients = req.body.ingredients.split(',');
+        const arrObjIngredients = [];
+        arrIngredients.forEach((ingredient) => {
+          Ingredient.findOrCreate({
+            where: {
+              name: ingredient,
+            },
+          })
+            .spread((objIngredient) => {
+              newRecipe.addIngredient(objIngredient);
+            })
+            .catch(next);
+        });
+        res.status(201).json(newRecipe);
+      } else {
```

Comment:
add a then block:

make sure you pass in the array of Ing promises and the newRecipe

newRecipe.addIngredients(Array of Ing)